src/main/java/org/jvnet/hudson/update_center/Plugin.java

```diff
@@ -460,7 +465,35 @@ public JSONObject toJSON() throws Exception {
 
         json.put("wiki", "https://plugins.jenkins.io/" + artifactId);
 
-        json.put("labels", getLabels());
+        GitHubSource gh = GitHubSource.getInstance();
+        ArrayList<String> labels = new ArrayList<String>();
+        labels.addAll(Arrays.asList(getLabels()));
+
+        if (scm != null && scm.contains("https://github.com/")) {
+            String[] parts = scm.replaceFirst("https://github.com/", "").split("/");
+            if (parts.length >= 2) {
+                labels.addAll(
+                    Arrays.asList(
+                        gh.getTopics(parts[0], parts[1]).toArray(new String[0])
+                    )
+                );
+            }
+        }
+
+        if (!labels.isEmpty()) {
+            HashSet<String> allowedLabels = new HashSet<String>();
+
+            for (String label : labels) {
+                // Everything starting with jenkins- is allowed
+                if (label.startsWith("jenkins-")) {
+                    allowedLabels.add(label.replaceFirst("jenkins-", ""));
+                } else if (ALLOWED_LABELS.contains(label)) {
+                    allowedLabels.add(label);
+                }
+            }
+
+            json.put("labels", allowedLabels);
```

Comment:
This is only executed when labels are coming from GitHub. Otherwise there will not be a 'labels' element.

Reply:
470 adds all the wiki/properties file to the labels array 

483 is supposed to be whitelist, and deleting jenkins- prefix

So it should handle both no?

Reply:
There are two problems here:

* First, if there are labels from the static file, they are completely ignored since this adds `allowedLabels`, not `labels`.
* Second, if there are _no_ labels from the static file, this call isn't done and the labels element is simply missing.

Reply:
Oh I can force the label to be there. That was probably oversight 

Labels list is Union of GitHub and static file. If both are empty your right, it won't run. I'll fix that. If either or both is non empty it should run, though the whitelisting is broken and needs fixing 

So tomorrow I'll fix the whitelisting logic, and force the labels element to always be there.